backend/svirt.pm

```diff
@@ -174,6 +176,32 @@ sub load_snapshot ($self, $args) {
     return $post_load_snapshot_command;
 }
 
+sub do_extract_assets ($self, $args) {
+    die "No name given" unless $args->{name};
+    die "No HDD specified" unless $args->{hdd_num};
+    die "No dir specified" unless $args->{dir};
+    die "No format specified" unless $args->{format};
+
+    my $format = $args->{format};
+    return undef if (($format ne 'raw') and ($format ne 'qcow2'));
+
+    my $first_hdd = $bmwqemu::vars{S390_ZKVM} ? 'a' : 'b';
+    my $name = $args->{name};
+    my $hdd_num = $args->{hdd_num} - 1;
+    my $vmname = $self->console('svirt')->name;
+    my $svirt_name = $vmname . chr(ord($first_hdd) + $hdd_num);
+    my $img_dir = $args->{dir};
+    mkpath($img_dir);
+    my $svirt_img_name = "$img_dir/$svirt_name.img";
```

Comment:
This is the place where the img file is stored, but in `$args->{dir}` we are getting the path where the final file should be stored.
Compare the original code https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/shutdown/svirt_upload_assets.pm#L22
```
    my $image_storage = '/var/lib/libvirt/images';
    my $svirt_img_name = $image_storage . '/' . $args->{svirt_name} . '.img';

    my $cmd = "nice ionice qemu-img convert -p -O $format $svirt_img_name $image_storage/$name";

    upload_asset("$image_storage/$name", 1, 1);
```

Reply:
Right. I was/am trying to match `qemu.pm` which only calls `export_blockdev_images` which uses `qemu-img` with `$args->{dir}` as the location... there's no upload code there as far as I can tell. I do have a feeling I might be missing something.

Reply:
In my test I'm checking the expected qemu-img command. `t/18-qemu` doesn't test `do_extract_assets` as such afair. Ideally I'd know what needs to be unit tested, to then make it do the right thing.

Reply:
I considered using `OpenQA::Qemu::Proc` directly, but that seems to require the drives and block devices... only to generate the qemu-img command in the end - although I'm leaning towards extracting `Qemu/DriveDevice::gen_qemu_img_convert` which then could be shared?

Reply:
I'll try to write down what I mean:
```perl
my $image_storage = '/var/lib/libvirt/images'; # maybe there is already a var or constant we can use, though
my $target_dir = $args->{dir}; # will contain the value 'assets_public'
my $svirt_img_name = "$image_storage/$svirt_name.img"; # this is where the source file is located
my $cmd = "nice ionice qemu-img convert -p -O $format $svirt_img_name $target_dir/$name"; # result file will end up in 'assets_public'
```

Reply:
I introduced a constant `IMAGE_STORAGE` so we don't need to hard-code this folder. And using `dir` for the target should be fine, since the comment in the distri code saying `private asset` is not what the code does.